source/material_model/rheology/diffusion_creep.cc

```diff
@@ -91,11 +105,34 @@ namespace aspect
         // A: prefactor,
         // d: grain size, m: grain size exponent, E: activation energy, P: pressure,
         // V; activation volume, R: gas constant, T: temperature.
-        double viscosity_diffusion = 0.5 / p.prefactor *
-                                     std::exp((p.activation_energy +
-                                               pressure*p.activation_volume)/
-                                              (constants::gas_constant*temperature)) *
-                                     std::pow(grain_size, p.grain_size_exponent);
+
+	double viscosity_diffusion = 0;
+
+	if (use_water_fugacity_diffusion)
+	  {
+	     AssertThrow(this->introspection().compositional_name_exists("water"),
+                         ExcMessage("The water fugacity rheology only works if "
+                                    "there is a compositional field called water."));
+            const unsigned int water_idx = this->introspection().compositional_index_for_name("water");
+            double water_content = this->introspection().component_indices.compositional_fields[water_idx];
+
+            viscosity_diffusion = 0.5 / (p.prefactor / std::pow(p.wet_prefactor, p.water_fugacity_exponent)) *
+		                         std::pow(water_content, -p.water_fugacity_exponent) *
```

Comment:
I'd suggest changing the sign of `p.water_fugacity_exponent`. It makes little sense to raise water concentration to a negative of a fugacity exponent.

Reply:
Wouldn't this require inputting a negative water_fugacity_exponent in the parameter file though? It seems like mathematically it'll do the same thing but it'll be more confusing when inputting values for models

Reply:
Mea culpa, I misread the equation. It's common to group the fugacity term with the prefactor (which is already part of the denominator of the first term), and if you did that, the negative sign would go away.

Reply:
> I updated the code and addressed your comments @bobmyhill but at the moment the method John and I talked about required adding an argument to compute_viscosity that I hardcoded to be 1 inside rheology/composite_visco_plastic.cc and diffusion_dislocation.cc, definitely not ideal but thought it would be useful to push anyway to see if you had any thoughts on an approach

I agree that it isn't a beautiful solution, but I don't think it's a bad choice. The alternative would be to make a separate "wet diffusion" rheology, but that doesn't seem any better.